### PR TITLE
gulp script properly combine streams in tasks

### PR DESCRIPTION
**Pull request by julienV**
see https://github.com/gulpjs/gulp/blob/master/docs/recipes/using-multiple-sources-in-one-task.md

### CODE REVIEW

File: build/gulp-redcore/components/redcore.js [hunk 1]
```diff
@@ -44,14 +45,13 @@ gulp.task('copy:' + baseTask,
...
 gulp.task('copy:' + baseTask + ':backend', ['clean:' + baseTask + ':backend'], function(cb) {
-	return (
-		gulp.src(extPath + '/admin/**')
-		.pipe(gulp.dest(config.wwwDir + '/administrator/components/com_redcore')) &&
-		gulp.src(extPath + '/../../redcore.xml')
-		.pipe(gulp.dest(config.wwwDir + '/administrator/components/com_redcore')) &&
-		gulp.src(extPath + '/../../install.php')
-		.pipe(gulp.dest(config.wwwDir + '/administrator/components/com_redcore'))
-	);
+	var admin = gulp.src(extPath + '/admin/**')
```
Comment (Kixo) [inline build/gulp-redcore/components/redcore.js]: you are missing redcore.xml here...

Comment (julienV) [inline build/gulp-redcore/components/redcore.js]: see line 51

**Comment by phproberto** [comment]
while this is correct you can do that just adding additional tasks:

```gulp.task('copy:' + baseTask + ':backend', [
    'copy:' + baseTask + ':backend-files',
    'copy:' + baseTask + ':installer',
], function(cb) {
});
```

then each task will take care of doing the specific thing you want.

**Comment by julienV** [comment]
sure you can, but maybe it's better that it stays on same task, which is 'moving files in admin folder'.
anyway, it was just to point out that indeed one task need to return one stream for proper scheduling if nothing else.

**Comment by phproberto** [comment]
Yes. But having a task is better because you can have a watch task for the installer. When the installer is modified only the installer task will be fired which will save time. Plus you save a dependency.

**Comment by phproberto** [comment]
Maybe your solution is better for this case because when you run copy:backend it will run clean:backend which may conflict with a clean:installer because the installer resides on the backend component folder. So it may try to delete/copy files from an existing folder

**Pull request merged**